pkg/importer/importer_test.go

```diff
@@ -0,0 +1,127 @@
+package importer_test
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+
+	. "github.com/kubevirt/containerized-data-importer/pkg/importer"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+type fakeDataStream struct {
+	url         *url.URL
+	accessKeyId string
+	secretKey   string
+	err         error
+}
+
+//var _ DataStreamInterface = &fakeDataStream{}
+
+func (d *fakeDataStream) Error() error {
+	return d.err
+}
+
+func (d *fakeDataStream) DataStreamSelector() (io.ReadCloser, error) {
+	if d.err != nil {
+		return nil, d.err
+	}
+	if d.url.Scheme == "s3" {
+		return d.s3()
+	}
+	return d.http()
+}
+
+func (d *fakeDataStream) s3() (io.ReadCloser, error) {
+	return ioutil.NopCloser(bytes.NewReader([]byte("s3 dataStream"))), nil
+}
+
+func (d *fakeDataStream) http() (io.ReadCloser, error) {
+	return ioutil.NopCloser(bytes.NewReader([]byte("http dataStream"))), nil
+}
+
+// parseDataPath only for debug, never used in mock
+func (d *fakeDataStream) parseDataPath() (string, string, error) {
+	return "", "", nil
+}
+
+// NewFakeDataStream: construct a new fakeDataStream object from params.
+func NewFakeDataStream(ep, accKey, secKey string) *fakeDataStream {
+	epUrl, err := url.Parse(ep)
+	if err != nil {
+		return &fakeDataStream{err: err}
+	}
+	return &fakeDataStream{
+		url:         epUrl,
+		accessKeyId: accKey,
+		secretKey:   secKey,
+	}
+}
+
+var _ = Describe("Importer", func() {
+	Context("Test StreamDataToFile when", func() {
+		var dataStream io.ReadCloser
+
+		const importerTestFolder = "/tmp/importer-test/"
+
+		BeforeEach(func() {
+			// create a files and importerTestFolder
+			var err error
+			if _, err = os.Stat(importerTestFolder); os.IsNotExist(err) {
+				os.Mkdir(importerTestFolder, os.ModePerm)
+			}
+			Expect(os.Create(importerTestFolder + "test_file_already_exist")).ToNot(BeNil())
+		})
+
+		It("use http", func() {
+			var err error
+			dataStream, err = NewFakeDataStream("http://www.google.com", "", "").DataStreamSelector()
+			Expect(err).ToNot(HaveOccurred())
+			StreamDataToFile(dataStream, importerTestFolder+"test-http")
```

Comment:
since `importerTestFolder+"test-http"` is used twice consider making it its own var.